controller/controller_test.go

```diff
@@ -386,73 +380,105 @@ func TestUpdaterIsUpdatedOnK8sUpdates(t *testing.T) {
 
 		updater.On("Start").Return(nil)
 		updater.On("Stop").Return(nil)
-		// once for ingress update, once for service update
-		updater.On("Update", test.entries).Return(nil).Times(2)
+		updater.On("Update", mock.MatchedBy(func(u IngressUpdate) bool {
+			assert.Len(u.Entries, len(test.expectedUpdate.Entries), "Expected same number of entries in update")
+			for _, entry := range test.expectedUpdate.Entries {
+				assert.Contains(u.Entries, entry, "Expected entry in update")
+			}
+			return true
+		})).Return(nil)
 
 		client.On("GetIngresses").Return(test.ingresses, nil)
-		client.On("GetServices").Return(test.services, nil)
+		client.On("GetEndpoints").Return(test.endpoints, nil)
 
-		ingressWatcher, ingressCh := createFakeWatcher()
-		serviceWatcher, serviceCh := createFakeWatcher()
+		ingressWatcher, _ := createFakeWatcher()
+		endpointsWatcher, endpointsCh := createFakeWatcher()
 		client.On("WatchIngresses").Return(ingressWatcher)
-		client.On("WatchServices").Return(serviceWatcher)
+		client.On("WatchEndpoints").Return(endpointsWatcher)
 
 		//when
 		assert.NoError(controller.Start())
-		ingressCh <- struct{}{}
-		serviceCh <- struct{}{}
+		endpointsCh <- struct{}{}
 		time.Sleep(smallWaitTime)
 
 		//then
 		assert.NoError(controller.Stop())
 		time.Sleep(smallWaitTime)
 		updater.AssertExpectations(t)
+
+		if t.Failed() {
```

Comment:
It won't fail until all the examples finish in the test function. This makes it much easier to see which example failed.